### ISSUE TITLE
require_ansible_source currently broken

### ISSUE THREAD

**Issue by antimius** [neillturner/kitchen-ansible#126]
Using require_ansible_source: true gives me the following error:

```
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #converge action: [undefined local variable or method `install_epel_repo' for #<Kitchen::Provisioner::AnsiblePlaybook:0x000000017c37a0>]
```

Also, is there a reason 'pip install ansible' as an install method is undesirable?

**Comment by neillturner** [comment]
i think this could only happen if the code didn't recognize the platform in this code in os.rb

def self.make(platform, config)
          return nil if platform == ''

          case platform
          when 'debian', 'ubuntu'
            return Debian.new(platform, config)
          when 'redhat', 'centos', 'fedora'
            return Redhat.new(platform, config)
          when 'amazon'
            return Amazon.new(platform, config)
          when 'suse', 'opensuse', 'sles'
            return Suse.new(platform, config)
          end

          nil
        end
as a workaround try setting in your .kitchen.yml file
   ansible_platform=centos   (or what your os is) 

1. I think the problem is that when PR Split installation scripts into their own classes #110
was done it didn't do the case of building from source so i will correct this. 

2. i'll add some messages to this code in os.rb  to make it more defensive. 
to say when no platform is matched.

**Comment by neillturner** [comment]
in regard to 'pip install ansible' my aim is to put as many install methods as possible for maximum flexibility particularly if they are contributed by others. -)
 so if you specify your requirement  in more detail. i.e. what platform etc or even better create a pull request i'm happy to add another install option.

**Comment by neillturner** [comment]
Actually the fix is to change in ansible_playbook.rb
in
def install_ansible_from_source_command

change 
#{install_epel_repo}
to            
#{Kitchen::Provisioner::Ansible::Os::Redhat.new('redhat', config).install_epel_repo}
change 
#{update_packages_redhat_cmd}
to 
#{Kitchen::Provisioner::Ansible::Os::Redhat.new('redhat', config).update_packages_redhat_cmd}
change 
#{update_packages_suse_cmd}
to
#{Kitchen::Provisioner::Ansible::Os::Redhat.new('redhat', config).update_packages_suse_cmd}
change 
#{update_packages_debian_cmd}
to 
#{Kitchen::Provisioner::Ansible::Os::Redhat.new('redhat', config).update_packages_debian_cmd}
I'll get this out and add a test case.

**Comment by neillturner** [comment]
closed - resolved in version 0.0.37
